src/bun.js/bindings/bindings.cpp

```diff
@@ -3944,7 +3944,10 @@ JSC::EncodedJSValue JSC__JSValue__getIfPropertyExistsImpl(JSC::EncodedJSValue JS
     const auto propertyString = String(StringImpl::createWithoutCopying({ arg1, arg2 }));
     const auto identifier = JSC::Identifier::fromString(vm, propertyString);
     const auto property = JSC::PropertyName(identifier);
-
+    if (auto index = parseIndex(property)) {
```

Comment:
This will incur a performance hit on every single property access from native code. We should not do this. Instead, we should only use this function when the value is known to be a string, and it should assert that it's not an integer.